spec.html

```diff
@@ -31678,20 +31542,147 @@
 
       <!-- es6num="22.2.4.1" -->
       <emu-clause id="sec-typedarray">
-        <h1>_TypedArray_( ... _argumentsList_)</h1>
-        <p>A _TypedArray_ constructor with a list of arguments _argumentsList_ performs the following steps:</p>
+        <h1>_TypedArray_ ( )</h1>
+        <p>This description applies only if the _TypedArray_ function is called with no arguments.</p>
         <emu-alg>
           1. If NewTarget is *undefined*, throw a *TypeError* exception.
-          1. Let _here_ be the active function.
-          1. Let _super_ be ? _here_.[[GetPrototypeOf]]().
-          1. If IsConstructor (_super_) is *false*, throw a *TypeError* exception.
-          1. Let _argumentsList_ be the _argumentsList_ argument of the [[Construct]] internal method that invoked the active function.
-          1. Return Construct(_super_, _argumentsList_, NewTarget).
+          1. Return AllocateTypedArray(_TypedArray_, 0).
+        </emu-alg>
+      </emu-clause>
+
+      <!-- es6num="22.2.1.2" -->
+      <emu-clause id="sec-typedarray-length">
+        <h1>_TypedArray_ ( _length_ )</h1>
+        <p>This description applies only if the _TypedArray_ function is called with at least one argument and the Type of the first argument is not Object.</p>
+        <p>_TypedArray_ called with argument _length_ performs the following steps:</p>
+        <emu-alg>
+          1. Assert: Type(_length_) is not Object.
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. If _length_ is *undefined*, throw a *TypeError* exception.
+          1. Let _numberLength_ be ToNumber(_length_).
+          1. Let _elementLength_ be ? ToLength(_numberLength_).
+          1. If SameValueZero(_numberLength_, _elementLength_) is *false*, throw a *RangeError* exception.
+          1. Return AllocateTypedArray(_TypedArray_, _elementLength_).
+        </emu-alg>
+
+        <!-- es6num="22.2.1.2.1" -->
+        <emu-clause id="sec-allocatetypedarray" aoid="AllocateTypedArray">
+          <h1>Runtime Semantics: AllocateTypedArray (_typedArray_, _length_ )</h1>
+          <p>The abstract operation AllocateTypedArray with argument _typedArray_ and optional argument _length_ is used to validate and create an instance of a TypedArray constructor. _typedArray_ is required to be one of the intrinsic TypedArray constructors which _TypedArray_ ranges over. If the _length_ argument is passed an ArrayBuffer of that length is also allocated and associated with the new TypedArray instance. AllocateTypedArray provides common semantics that is used by all of the _TypeArray_ overloads and other methods. AllocateTypedArray performs the following steps:</p>
+          <emu-alg>
+            1. Let _proto_ be the initial value of the [[Prototype]] internal slot of _typedArray_.
```

Comment:
I think you actually want to pass "NewTarget" to AllocateTypedArray and then use "GetPrototypeFromConstructor(NewTarget)", otherwise `class I8Array extends Int8Array {}` doesn't use `I8Array.prototype`.